flink-clients/src/main/java/org/apache/flink/client/CliFrontend.java

```diff
@@ -588,15 +653,16 @@ protected int cancel(String[] args) {
 			ActorGateway jobManager = getJobManagerGateway(options);
 			Future<Object> response = jobManager.ask(new CancelJob(jobId), askTimeout);
 
-			try {
-				Await.result(response, askTimeout);
-				return 0;
-			}
-			catch (Exception e) {
-				throw new Exception("Canceling the job with ID " + jobId + " failed.", e);
+			Object rc = Await.result(response, askTimeout);
+
+			if (rc instanceof CancellationFailure) {
```

Comment:
Maybe we can do the same with the `CancellationFailure` as with the `StopFailure`, because I haven't found a place where we use the `jobID` field of the `CancellationFailure` instance.